sycl/test-e2e/Matrix/joint_matrix_apply_two_matrices_impl.hpp

```diff
@@ -70,22 +76,32 @@ bool apply_two_matrices(Tc *C, Tc *D, Ta *A, Ta *Ar, queue q) {
            joint_matrix_load(
                sg, sub_c, pC + (sg_startx * TM) * N + sg_starty / sg_size * TN,
                N, layout::row_major);
-           joint_matrix_apply(sg, sub_c, sub_d,
-                              [](const Tc &x, Tc &y) { y = x * 2; });
+           joint_matrix_apply(sg, sub_c, sub_d, [](Tc &x, Tc &y) {
+             y = x * 2;
```

Comment:
Each of these 4 math operations inside lambda are duplicated in 2 places. Let's move them to simple functions like:
```
template<typename T>
T add2(T a) {
  return a + 2;
}
```
Also to simplify test, can we use the same operation, not 4 different? Then each apply would contain this:
```
y = add2(x);
x = add2(x);
```